### PR TITLE
fix: added logic and styles to confirm view

### LINKED ISSUE #43: Add Confirm view Page
# Description

When DB state for Ticket changes to `delivery_confirmed` a new view changes on the User's App. We see Title, Subtitle, Icon/Image, a Form that has a Title, 3 inputs, and a Button:

| Mobile | Desktop |
| --- | --- |
| image | image |

Notice:
- Same problems as Check view, as noted below. Fixing one should apply to the other
- Layout elements (let's call it BLUE and BG) changes a lot. BLUE loses the border-radius and BG changes color! Also, the title of the form changes color
- Image changes align from start to end in her div
- Back button gains a Text (why not have it across all sizes?
- This should show input errors (ticket doesn't exist on DB, format not allowed, others TBD)
- This should show error toast for server errors (server unavailable, server error), others TBD), or Client High Severity errors.
- Should Button be disabled if no input answer is given? Are all 3 inputs required?
- Should Button have a `sending` state?

Clicking Button changes Ticket state to `user_confirmed`

Clicking Button opens a Modal for rating Delivery:

image

Notice:
- Close Button is small and color pale
- Stars are close to the Button so can cause problems of users clicking the button when wanting to choose Star
- Should Button be disabled if no Rate is given?
- Should the button have a `sending` state?

Clicking the button sends the rating to DB and change the Ticket state to `rated`

Clicking Button opens a Modal for Finishing flow:

image

Notice:
- Icon/Image is before Title
- What do close icons do? Closes modal and restart modal flow? Or redirects to Home?

Clicking the button should redirect to the Home view

## Checklist

- [x] Create Confirm View Page
- [ ] Confirm View Page is showing when ticket change for `delivery_confirmed`
- [ ] Clicking Button changes Ticket state to `user_confirmed` and open a modal for rating delivery
- [ ] Clicking Rating Modal Button sends the rating to DB and change the Ticket state to `rated`
- [ ] To the end is shown a modal to finishing flow

Related issues

- #21
- #30
- #45 
- #46

### PR DESCRIPTION
**Pull request by denisosuna**
### Description

Was added:

- the query to confirm the form
- the hook to manage the query
- the implementation of the insertRating hook
- the logic to catch the selected rating
- styles to the confirm view Page

Note: some confirm view page styles need to be fixed
fixes #43 

#### Screenshots
image

#### Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

### How to test it

If applicable, describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

### Checklist:

- [ ] Doesn't break the current code.
- [ ] Passes linters and test, also is building.
- [ ] Doesn't have spelling or grammatical problems.
- [ ] Doesn't have unnecessary comments or debugging code.
- [ ] The branch is updated with the last dev branch changes.

### CODE REVIEW

File: src/components/Rating/Rating.jsx [hunk 1]
```diff
@@ -1,14 +1,17 @@
 import { StarBtn, SelectedStarBtn } from './Rating.styled'
 import { useState } from 'react'
-
-const Rating = () => {
+import PropTypes from 'prop-types'
+// @ts-check
+// eslint-disable-next-line react/prop-types
```
Comment (heliomar-pena) [inline src/components/Rating/Rating.jsx]: Theses comments are required or your have missed delete it?

File: src/pages/Confirm/Confirm.jsx [hunk 2]
```diff
@@ -1,49 +1,62 @@
...
+      confirmPackage(event, id)
+    } else {
+      alert('Por favor complete los campos')
+    }
   }
 
   return (
     <Layout
       headerTitle="Hemos terminado"
+      headerSubTitle="Inserta tus datos para finalizar Confirma recepción"
```
Comment (heliomar-pena) [inline src/pages/Confirm/Confirm.jsx]: This paragraph is ok?

I ask for this part: `finalizar Confirma recepción`

**Comment by elpollo12** [comment]
I am adding the remaining styles

**Review by heliomar-pena — CHANGES_REQUESTED** [review]
Good job, some questions

File: src/components/Rating/Rating.jsx (hunk 1 above)
Comment (Israel-Laguan) [reply src/components/Rating/Rating.jsx]: I fixed that

**Pull request merged**